server/message_queue_service.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import json
+from typing import Dict
+
+import aio_pika
+from aio_pika import DeliveryMode, ExchangeType
+from aio_pika.exceptions import ProbableAuthenticationError
+
+from .config import config, TRACE
+from .core import Service
+from .decorators import with_logger
+
+
+@with_logger
+class MessageQueueService(Service):
+    def __init__(self) -> None:
+        """
+        Service handling connection to the message queue
+        and providing an interface to publish messages.
+        """
+        self._logger.debug("Message queue service created.")
+        self._connection = None
+        self._channel = None
+        self._exchanges = {}
+        self._exchange_types = {}
+
+        config.register_callback("MQ_USER", self.reconnect)
+        config.register_callback("MQ_PASSWORD", self.reconnect)
+        config.register_callback("MQ_VHOST", self.reconnect)
+        config.register_callback("MQ_PORT", self.reconnect)
+
+    async def initialize(self) -> None:
+        if self._connection is not None:
+            return
+
+        await self._connect()
+
+    async def _connect(self) -> None:
+        try:
+            self._connection = await aio_pika.connect_robust(
+                "amqp://{user}:{password}@localhost:{port}/{vhost}".format(
```

Comment:
Are there supposed to be two slashes between port and `vhost`? The `MQ_VHOST` variable starts with a slash.

Reply:
Indeed it's weird, but that's what was done for `/faf-postal` so I did the same. 
https://github.com/FAForever/faf-stack/blob/b6582560c77ddce549c0e099b1265b0d481c6805/config.template/faf-postal/faf-postal.env#L11